include/geode/image/core/raster.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2019 - 2023 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#pragma once
+
+#include <geode/basic/array.h>
+#include <geode/basic/identifier.h>
+#include <geode/basic/pimpl.h>
+
+#include <geode/image/common.h>
+
+namespace geode
+{
+    class RGBColor;
+} // namespace geode
+
+namespace geode
+{
+    template < index_t dimension >
```

Comment:
same here... what is a raster? 

we should clarify to avoid confusion with the file rasterize... 
We can think that algorithm in the file ratserize create a Raster but it does not.

Reply:
Raster -->PixelGrid and Array --> GridBase?

Reply:
https://en.wikipedia.org/wiki/Raster_graphics
raster is the scientific name for a pixel based image.

Reply:
renamed into RasterImage